api/admin/routes.py

```diff
@@ -48,20 +45,6 @@ def setup_admin(_db=None):
     app.secret_key = ConfigurationSetting.sitewide_secret(
         _db, Configuration.SECRET_KEY
     )
-    # If an admin was logged in when the app restarted and
-    # makes the first request, the flask session is loaded
-    # before this method. Since the secret key isn't set yet
-    # it ends up being a NullSession. We can't replace
-    # the flask session object, but we can create a new
-    # session object in order to extract the admin's email
-    # and store it for later.
-    if not flask.session and flask.request:
-        temp_session = app.open_session(flask.request)
-        email = temp_session.get("admin_email")
-        type = temp_session.get("auth_type")
-        if email and type:
-            app.admin_email = email
-            app.admin_auth_type = type
```

Comment:
This turned out to cause problems when you can log out, it sometimes kept using these variables on subsequent requests. I don't have another fix for it, so now you may get a 401 on the first request after the app reloads again.